tests/upsert_validator/test_validator_election.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from bigchaindb.upsert_validator import ValidatorElection
+from bigchaindb.common.exceptions import DuplicateTransaction
+
+pytestmark = [pytest.mark.tendermint, pytest.mark.bdb]
+
+
+def test_upsert_validator_valid_election(b_mock, new_validator, node_key):
+    voters = ValidatorElection.recipients(b_mock)
+    election = ValidatorElection.generate([node_key.public_key],
+                                          voters,
+                                          new_validator, None).sign([node_key.private_key])
+    assert election.validate(b_mock)
+
+
+def test_upsert_validator_invalid_election(b_mock, new_validator, node_key):
+    voters = ValidatorElection.recipients(b_mock)
+    valid_election = ValidatorElection.generate([node_key.public_key],
+                                                voters,
+                                                new_validator, None).sign([node_key.private_key])
+
+    with pytest.raises(DuplicateTransaction):
+        valid_election.validate(b_mock, [valid_election])
+
+    b_mock.store_bulk_transactions([valid_election])
+
+    with pytest.raises(DuplicateTransaction):
```

Comment:
Looks like these two `with pytest.raises(DuplicateTransaction):` are following the same code path.

Reply:
Not quite, the previous exception will be generated because `valid_election` is a part of `current_transactions` list and the second exception would be generated because `valid_election` has been stored in MongoDB and should be read from there to raise the exception.

Reply:
I think, you just need to call `valid_election.validate(b_mock)` in the second case. Otherwise you depend on the order of the checks.

Reply:
oh yeah, my bad.